windows/windows_online_updates_install/windows_online_updates_install.yml

```diff
@@ -0,0 +1,87 @@
+# Copyright 2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Description:
+#   This test case is used for installing Windows online updates in
+# guest OS through Windows Update. If parameter 'windows_updated_build_num'
+# is not set, this test case will try to search available Windows online updates
+# and install without checking updates categories, updated OS build number.
+#
+- name: windows_online_updates_install
+  hosts: localhost
+  gather_facts: no
+  tasks:
+    - name: "Test case block"
+      block:
+        - name: "Initialize Windows updates install variables"
+          ansible.builtin.set_fact:
+            win_updates_log_file: "C:\\win_updates_log.txt"
+            win_online_updates_installed: false
+            win_check_is_updated: false
+
+        - name: "Check parameter 'windows_updated_build_num' value"
+          ansible.builtin.assert:
+            that:
+              - windows_updated_build_num is regex("^[0-9]{5}(\.[0-9]+)?$")
+            fail_msg: >-
+              Parameter 'windows_updated_build_num' is not set to a valid value: "{{ windows_updated_build_num }}",
+              please set it to a valid build number, e.g., '22000', '22000.1', '22000.1234'.
+          when:
+            - windows_updated_build_num is defined
+            - windows_updated_build_num
+
+        - name: "Test setup"
+          include_tasks: ../setup/test_setup.yml
+
+        - name: "Get guest OS version build before installing Windows updates"
+          include_tasks: ../utils/win_get_os_version.yml
+        - name: "Display guest OS version build before installing Windows updates"
+          ansible.builtin.debug: var=win_os_version_build
+        - name: "Set fact of guest OS build number before installing Windows updates"
+          ansible.builtin.set_fact:
+            win_os_build_before_update: "{{ ('.').join(win_os_version_build.split('.')[2:]) }}"
+
+        # windows_updated_build_num should be set to the same or newer build num
+        # than the current one
+        - name: "Check parameter 'windows_updated_build_num' value"
+          ansible.builtin.assert:
+            that:
+              - windows_updated_build_num is version(win_os_build_before_update, '>=')
```

Comment:
If `windows_updated_build_num` == `win_os_build_before_update`, shall this case be skipped? It seems meaningless to "upgrade" same Windows build

Reply:
This is just an allowed configuration value, updates can be installed but build number not increased.